### PR TITLE
JOSS paper

### PR DESCRIPTION
**Pull request by zingale**
## PR summary

This is the JOSS paper for Castro.

## PR checklist

- [ ] test suite needs to be run on this PR
- [ ] this PR will change answers in the test suite
- [ ] all functions have docstrings as per the coding conventions
- [ ] the `CHANGES` file has been updated
- [ ] if appropriate, this change is described in the docs

### CODE REVIEW

**Comment by zingale** [comment]
Castro developers: I have populated the author list of this based on the current .zenodo.json file.  If you have contributed to Castro and believe you should be a co-author, please comment here or email me directly.  

If you do not wish to be part of the JOSS paper, likewise comment or issue a PR removing yourself.

Also please take the time to go over the paper and push changes as you see fit.  Once we converge, I'll open this PR for review and we can merge and then work on the JOSS submission.

**Comment by maxpkatz** [comment]
The JOSS standards say that the paper should include:

> A summary describing the high-level functionality and purpose of the software for a diverse, non-specialist audience.

I am concerned that the summary as written is a little too technical for a non-specialist audience. There are comments like "Astrophysical reaction networks are stiff" (which would require one to understand that a reaction network is a set of coupled ODEs to parse correctly) and "Castro includes full self-gravity with isolated boundary conditions" that I think are more aimed at astrophysicists than general software developers. So I suggest we take a pass through the paper and decide what specific methodology is crucial to describe and which parts we can leave out.

File: paper/paper.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+the early development.  The large number of application codes using
+AMReX means that Castro continually gains optimizations for new
+architectures.  As Castro evolved, we adopted a fully open development
+model (as does the Enzo [@enzo] code, for example).  We pride ourselfs in
+making all of the science problems available Castro git repository as
+we are developing them--no separate repo is used for our science work.
+Other simulation codes, like Flash [@flash], work with a general equation of
+state and reaction network, but Castro is unique in focusing on
+spectral deferred correction techniques for coupling the hydro and
+reactions.  Finally, while some astrophysics codes has forks that
```
Comment (harpolea) [inline paper/paper.md:120]:
```suggestion
reactions.  Finally, while some astrophysics codes have forks that
```

File: paper/paper.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+MPI to distribute AMR grids across nodes and using logical tiling with
+OpenMP to divide a grid across threads for manycore machines or CUDA
+to spread the work across GPU threads on GPU-based machines.  All of
+the core physics can run on GPUs (CTU hydro, CT-CTU MHD, self-gravity,
+diffusion, reactions) and has been shown to scale well to 1000s of
+GPUs [@castro_2019] and 100,000s of CPUs [@castro_2017].  For
+performance portability, the same compute kernel is used for CPUs or
+GPUs, using lambda-capturing for C++ kernels.
+
+While there are a number of astrophysical simulation codes, Castro
```
Comment (harpolea) [inline paper/paper.md:105]:
```suggestion
While there are a number of astrophysical hydrodynamics simulation codes, Castro
```

File: paper/paper.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+offers a few unique features.  The original motivation for developing
+Castro was to build a simulation code based on a modern,
+well-supported AMR library (BoxLib which evolved to AMReX), using
+unsplit integration techniques and targeting problems in nuclear
+astrophysics.  The radiation solver was a key design consideration in
+the early development.  The large number of application codes using
+AMReX means that Castro continually gains optimizations for new
+architectures.  As Castro evolved, we adopted a fully open development
+model (as does the Enzo [@enzo] code, for example).  We pride ourselfs in
```
Comment (harpolea) [inline paper/paper.md:114]:
```suggestion
model (as does the Enzo [@enzo] code, for example).  We pride ourselves in
```

File: paper/paper.md [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+making all of the science problems available Castro git repository as
```
Comment (harpolea) [inline paper/paper.md:115]:
```suggestion
making all of the science problems available in the Castro git repository as
```

**Comment by zingale** [comment]
I think I address all the comments

File: paper/paper.md (hunk 4 above)
Comment (harpolea) [reply paper/paper.md:115]: I think this still is missing a couple of words?

Comment (zingale) [reply paper/paper.md]: fixed

**Pull request merged**